controllers/goflowkube/goflowkube_reconciler.go

```diff
@@ -63,6 +69,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, desired *flowsv1alpha1.FlowC
 			newSVC := buildService(desired, r.OperatorNamespace)
 			r.createOrUpdate(ctx, oldSVC, newSVC, serviceKind)
 		}
+
+		//Delete or Create / Update Autoscaler according to HPA option
+		if desired.HPA == nil {
+			r.delete(ctx, oldASC, autoscalerKind)
+		} else if oldASC == nil || autoScalerNeedsUpdate(oldASC.(*ascv1.HorizontalPodAutoscaler), desired) {
+			newASC := buildAutoScaler(desired, r.OperatorNamespace)
+			r.createOrUpdate(ctx, oldASC, newASC, autoscalerKind)
+		}
 	case daemonSetKind:
```

Comment:
in the daemonset case we should certainly delete the HPA to not leave any unused object out there